mod/parse_url.php

```diff
@@ -1,516 +1,152 @@
 <?php
+
 /** 
  * @file mod/parse_url.php
+ * @brief The parse_url module
  * 
- * @todo https://developers.google.com/+/plugins/snippet/
- * 
- * @verbatim
- * <meta itemprop="name" content="Toller Titel">
- * <meta itemprop="description" content="Eine tolle Beschreibung">
- * <meta itemprop="image" content="http://maple.libertreeproject.org/images/tree-icon.png">
+ * This module does parse an url for embedable content (audio, video, image files or link)
+ * information and does format this information to BBCode or html (this depends
+ * on the user settings - default is BBCode output).
+ * If the user has enabled the richtext editor setting the output will be in html
+ * (Note: This is not always possible and in some case not useful because
+ * the richtext editor doesn't support all kind of html).
+ * Otherwise the output will be constructed BBCode.
  * 
- * <body itemscope itemtype="http://schema.org/Product">
- *   <h1 itemprop="name">Shiny Trinket</h1>
- *   <img itemprop="image" src="{image-url}" />
- *   <p itemprop="description">Shiny trinkets are shiny.</p>
- * </body>
- * @endverbatim
+ * @see ParseUrl::getSiteinfo() for more information about scraping embeddable content 
 */
 
-if(!function_exists('deletenode')) {
-	function deletenode(&$doc, $node)
-	{
-		$xpath = new DomXPath($doc);
-		$list = $xpath->query("//".$node);
-		foreach ($list as $child)
-			$child->parentNode->removeChild($child);
-	}
-}
-
-function completeurl($url, $scheme) {
-	$urlarr = parse_url($url);
-
-	if (isset($urlarr["scheme"]))
-		return($url);
-
-	$schemearr = parse_url($scheme);
-
-	$complete = $schemearr["scheme"]."://".$schemearr["host"];
+use \Friendica\ParseUrl;
 
-	if (@$schemearr["port"] != "")
-		$complete .= ":".$schemearr["port"];
+require_once("include/items.php");
 
-		if(strpos($urlarr['path'],'/') !== 0)
-			$complete .= '/';
-
-	$complete .= $urlarr["path"];
-
-	if (@$urlarr["query"] != "")
-		$complete .= "?".$urlarr["query"];
-
-	if (@$urlarr["fragment"] != "")
-		$complete .= "#".$urlarr["fragment"];
-
-	return($complete);
-}
-
-function parseurl_getsiteinfo_cached($url, $no_guessing = false, $do_oembed = true) {
```

Comment:
Can we split the addon repo in a master and in a develop branch? This is really annoying. We have the cache stuff which we need to change in addons, now we have this one. Do we have to change all this stuff at the day of release? This isn't a good workflow in my opinion.

We need a better solution